File: coj/1888-EvilNumbers.py
def containssix (number):
	while number > 0:
		if (number % 10) == 6:
			return True;
		number /= 10
	return False

def ismultipleofsix (number):
	return (number % 6) == 0

def sumismultipleofsix (number):
	sum = 0

	while number > 0:
		sum += number % 10
		number /= 10

	return ismultipleofsix(sum);

def isevil (number):
	return containssix(number) and ismultipleofsix(number) and sumismultipleofsix(number)

try:
    while True:
    	n = int(raw_input())
    	print "EVIL" if isevil(n) else "GOOD"
except EOFError:
    exit

File: coj/1888-EvilNumbers.cpp
#include <iostream>
typedef unsigned long long ll;

using namespace std;

bool containssix(ll number) {
	
	while ( number > 0) {
		if ((number % 10) == 6)
			return true;
		number /= 10;
	}
	return false;
}

bool ismultipleofsix(ll number) {

	return (number % 6) == 0;
}

bool sumismultipleofsix(ll number) {

	ll sum = 0;

	while ( number > 0) {
		sum += number % 10;
		number /= 10;
	}

	return ismultipleofsix(sum);
}

bool isevil(ll number) {

	return containssix(number) && ismultipleofsix(number) && sumismultipleofsix(number);
}

int main() {
	
	ll n;

	while((cin >>n), !cin.eof()) {
		cout << ((isevil(n))? ("EVIL"): ("GOOD")) << endl;
	}
}
